code-challenges.test.js

```diff
@@ -23,37 +23,132 @@ const numbersArray2 = [24, 27, 30, 33, 36]
+//     console.log(object)
+//     if (object.number % 3 === 0) {
+//         return `${object.number} is divisible by three`
+//     } else if (object % 3 !== 0) {
+//         return `${object.number} is not divisible by three`
+// }
+// }
+
+// divThree tested
+
+//  1 passed! Green!
+
+
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+// describe('capIt', () => {
+//     it("Takes in an array of words and returns an array with all the words capitalized", () => {
+//         const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+//         // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+//         const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+//         // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+//         expect(capIt(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+//         expect(capIt(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+//     })
+// })
+
+
+// ReferenceError: capIt is not defined
 
 // b) Create the function that makes the test pass.
 
+// Create a function named capIt that takes in the array and capitalize each strings first character.
+// This function will utilize methods such as .map(), .toUpperCase, and  newly found .substring (used so that the rest of the element does not get capitalized)
+// Return will bring back a new array with the index 0 capitalized only.
+
+// const capIt = (array) => {
+//     return array.map((value) => {
+//         return value[0].toUpperCase() + value.substring(1)
+//     })
+// }
+
+// Test Suites: 1 passed, 1 total
+
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
+
+
+describe('firstVowel', () => {
+    it('Takes in a string and logs the index of the first vowel', () => {
+        expect(firstVowel(vowelTester1)).toEqual(1)
+        expect(firstVowel(vowelTester2)).toEqual(0)
+        expect(firstVowel(vowelTester3)).toEqual(2)
+        const vowelTester1 = "learn"
+        // Expected output: 1
+        const vowelTester2 = "academy"
+        // Expected output: 0
+        const vowelTester3 = "challenges"
+        // Expected output: 2
```

Comment:
Best practice would be to have the variables declared just before the expect statement.